specification/_types/common.ts

```diff
@@ -24,10 +24,10 @@ import { AdditionalProperties } from '@spec_utils/behaviors'
 
 /**
  * A field value.
- * @codegen_names long, double, string, boolean
+ * @codegen_names long, double, string, boolean, user_defined_value
```

Comment:
`user_defined_value` feels a bit leaky here since these names are user-facing, should we use something else like `raw` or `any`?

Reply:
+1. `any` sounds good here.

It is to be noted that we could just now use `UserDefinedValue` everywhere instead of `FieldValue` but it has a specific semantics, and scalar values are used a lot more often than raw json. So having a dedicated `FieldValue` abstraction allows code language clients to use a specific type for it with convenience functions to create a `FieldValue` from simple scalar values.

BTW, the above could be captured in a code comment (and not a jsdoc comment) to explain the rationale for `FieldValue` :-)

Reply:
... and we can also remove the `FIXME` below since adding user defined value now handles geopoints and ip addresses!

Reply:
Agreed on `any`, I'll make the change!